### PR TITLE
Event Grid: Regenerated the SDK and re-recorded the tests based on the newer API version

### PR DESCRIPTION
**Pull request by kalyanaj**
## Description
OpenAPI Spec PR link: https://github.com/Azure/azure-rest-api-specs/pull/2239

This checklist is used to make sure that common guidelines for a pull request are followed.
- [X] Please add REST spec PR link to the SDK PR
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).** (Note: Per the feedback during the Swagger spec review, EventSubscription_Create has been renamed to EventSubscription_CreateOrUpdate. The service is still in Preview.).

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [X] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [X] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [X] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [X] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/EventGrid/management/Management.EventGrid/Microsoft.Azure.Management.EventGrid.csproj [hunk 1]
```diff
@@ -6,7 +6,7 @@
 	<PropertyGroup>
 		<PackageId>Microsoft.Azure.Management.EventGrid</PackageId>
 		<Description>Provides developers with a library to create and manage all Azure EventGrid resources. </Description>
-		<Version>1.1.0-preview</Version>
+		<Version>1.2.0-preview</Version>
```
Comment (shahabhijeet) [inline src/SDKs/EventGrid/management/Management.EventGrid/Microsoft.Azure.Management.EventGrid.csproj]: @kalyanaj update package release notes for this release in .csproj

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Plus fix failing test causing CI to fail

File: src/SDKs/EventGrid/management/Management.EventGrid/Microsoft.Azure.Management.EventGrid.csproj (hunk 1 above)
Comment (kalyanaj) [reply src/SDKs/EventGrid/management/Management.EventGrid/Microsoft.Azure.Management.EventGrid.csproj]: Done, thanks.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
It seems this particular version introduced breaking change in the public API surface.
Also this SDK is targeting new API version, it will be better if you make the version 2.0.0
This version change should happen at two places
Nuget version  2.0.0 (in .csproj)
Assembly/AssemblyFileVersion  2.0.0 (in AssemblyInfo.cs)

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**